Tests/SonarScanner.MSBuild.Shim.Test/ScannerEngineInputGeneratorTest.GenerateResult.cs

```diff
@@ -786,6 +787,76 @@ public void GenerateResult_AdditionalFiles_OnlyTestFiles_EndToEnd()
         CreateInputReader(result)["sonar.tests"].Split(',').Select(x => x.Trim('\"')).Should().BeEquivalentTo(testFiles);
     }
 
+    [TestMethod]
+    public void GenerateResult_CommandLineArgs_AddedToEngineInput()
+    {
+        var commandLineArgs = new ListPropertiesProvider
+        {
+            { SonarProperties.SonarPassword, "secret pwd" },
+            { SonarProperties.SonarUserName, "secret username" },
+            { SonarProperties.SonarToken, "secret token" },
+            { SonarProperties.ClientCertPassword, "secret client certpwd" },
+            { "sonar.some.other.arg", "someValue" }
+        };
+        var result = CreateInputReader(new ScannerEngineInputGenerator(CreateValidConfig(), runtime, commandLineArgs).GenerateResult());
```

Comment:
`new ScannerEngineInputGenerator(CreateValidConfig(), runtime, commandLineArgs).GenerateResult()` is `result`. You have `reader` here